src/components/SVGViewer/SVGViewer.tsx

```diff
@@ -0,0 +1,690 @@
+    zoomCenter?: ZoomCenter
+  ) => {
+    if (!this.pinchZoomInstance) {
+      return;
+    }
+    const startZoomFactor = this.pinchZoomInstance.zoomFactor;
+    const center = zoomCenter || {
+      x: this.pinchZoomInstance.container.clientWidth / 2,
+      y: this.pinchZoomInstance.container.clientHeight / 2,
+    };
+    const updateProgress = (progress: number) => {
+      this.pinchZoomInstance.scaleTo(
+        startZoomFactor + progress * zoomProgress,
+        center
+      );
+    };
+    this.pinchZoomInstance.animate(
+      this.pinchZoomInstance.options.animationDuration,
+      updateProgress,
+      this.pinchZoomInstance.swing
+    );
+    if (this.props.handleAnimateZoom) {
+      this.props.handleAnimateZoom({ zoomProgress, source, zoomCenter });
+    }
+  };
+
+  zoomOnCurrentAsset = (currentAsset: Element | null) => {
+    const isDesktop = this.state.width > 992;
+    if (!currentAsset || !this.pinchZoomInstance.container) {
+      return;
+    }
+    this.pinchZoomInstance.zoomFactor = isDesktop
+      ? zoomLevel * 3
+      : zoomLevel * 10;
+    // Need to wait until zoom applies to get proper offsets
+    setTimeout(() => {
+      const currentAssetPosition = currentAsset.getBoundingClientRect();
+      if (currentAssetPosition) {
+        this.pinchZoomInstance.offset = {
+          x:
+            currentAssetPosition.left -
+            this.pinchZoomInstance.container.clientWidth / 2 +
+            this.pinchZoomInstance.offset.x,
+          y:
+            currentAssetPosition.top -
+            this.pinchZoomInstance.container.clientHeight / 2 +
+            this.pinchZoomInstance.offset.y,
+        };
+        this.pinchZoomInstance.update();
+      }
+    });
+  };
+
+  zoomIn = () => {
+    this.animateZoom(zoomLevel, 'topbar');
+  };
+
+  zoomOut = () => {
+    if (!this.pinchZoomInstance) {
+      return;
+    }
+    let zoomFactor;
+    const startZoomFactor = this.pinchZoomInstance.zoomFactor;
+    if (startZoomFactor - zoomLevel > 1) {
+      zoomFactor = startZoomFactor - zoomLevel;
+    } else {
+      zoomFactor = 1;
+    }
+    this.animateZoom(zoomFactor - startZoomFactor, 'topbar');
+  };
+
+  openSearch = () => {
+    this.setState({ isSearchVisible: true });
+  };
+
+  handleCancelSearch = () => {
+    this.setState({
+      isSearchVisible: false,
+    });
+  };
+
+  handleCloseModal = () => {
+    if (this.props.handleCancel) {
+      this.props.handleCancel();
+    }
+  };
+
+  handleSearchFocus = () => {
+    this.setState({ isSearchFocused: true });
+  };
+
+  handleSearchBlur = () => {
+    this.setState({ isSearchFocused: false });
+  };
+}
+
+const SVGViewerContainer = styled.div`
+  height: 100%;
+  width: 100%;
+`;
+
+const SvgNode = styled.div`
+  cursor: move;
+  overflow: auto;
+  height: 100%;
+  width: 100%;
+  background: #ffffff;
+  > svg {
+    display: block;
+  }
+  .search-result {
+    &.metadata-container {
+      text {
+        stroke: #c945db !important;
```

Comment:
Please consider making an additional prop to redefine these default colors.

Reply:
same here, see below